src/shims/linux_stubs.c

```diff
@@ -52,9 +52,15 @@ char* mach_error_string(mach_msg_return_t x) {
 }
 void mach_vm_deallocate() { LINUX_PORT_ERROR();  }
 
+#if DISPATCH_USE_THREAD_LOCAL_STORAGE
+mach_port_t pthread_mach_thread_np(void) {
```

Comment:
not the right patch, you should instead re-define _dispatch_thread_port() to be the tid that you computed.

also __dispatch_tsd.tid may not be initialized, seems wrong.